core/sr-primitives/src/testing.rs

```diff
@@ -44,6 +44,12 @@ impl TypedKey for UintAuthorityId {
 	const KEY_TYPE: KeyTypeId = UINT_DUMMY_KEY;
 }
 
+impl AsRef<[u8]> for UintAuthorityId {
+	fn as_ref(&self) -> &[u8] {
+		unsafe { &std::mem::transmute::<_, &[u8; 4]>(&self.0)[..] }
```

Comment:
Let's be **extremely** careful when using transmute. This code definitely isn't correct.

Reply:
Besides `u64` being `[u8;8]`, I don't see any problem here?

Reply:
And casting to `[u8;4]` is valid.

Reply:
Yeah, my fault. We should use `slice::from_raw_parts` here to get a correct fat pointer.